rust/src/delta_datafusion.rs

```diff
@@ -163,9 +209,53 @@ impl TableProvider for delta::DeltaTable {
     }
 }
 
-fn stat_to_val(stat: &ColumnCountStat) -> Option<usize> {
-    match stat {
-        ColumnCountStat::Value(val) => Some(*val as usize),
-        ColumnCountStat::Column(_) => None,
+fn to_scalar_value(stat_val: &serde_json::Value) -> Option<datafusion::scalar::ScalarValue> {
+    if stat_val.is_number() {
+        if let Some(val) = stat_val.as_i64() {
+            Some(ScalarValue::from(val))
+        } else if let Some(val) = stat_val.as_u64() {
+            Some(ScalarValue::from(val))
+        } else {
+            stat_val.as_f64().map(ScalarValue::from)
+        }
+    } else {
```

Comment:
Probably some more easy to add are string / boolean.

Reply:
Yea, just wondering if max/min values of string/boolean are kept in statistics too.